prover/src/lib.rs

```diff
@@ -131,6 +136,12 @@ impl Prover for ExecutionProver {
             "provided outputs do not match the execution trace"
         );
 
-        PublicInputs::new(trace.program_hash(), self.stack_inputs.clone(), self.outputs.clone())
+        // TODO we override the kernel so the verification won't fail for CLI applications where we
+        // don't have an available kernel
+        let program_hash = *trace.program_info().program_hash();
+        let kernel = Kernel::default();
+        let program_info = ProgramInfo::new(program_hash, kernel);
```

Comment:
Here we do have the correct kernel via `trace.program_info()`, but we need to override it so it will correspond to the one used for the verification.

Once we have the CLI kernel argument, we can just get rid of this block and pass `trace.program_info()` as argument for `PublicInputs::new`

Reply:
Not sure I follow. We can't use kernels in the CLI either for proving or for verification - so, it should be the same in both cases. On the other hand, if we use prover and verifier programmatically, we can provide kernels and it should behave as expected.

I don't think we should tailor the code in `prover` and `verifier` crates for the needs of CLI as CLI is just one way to use the system.